gocd/bci_repo_publish.py

```diff
@@ -173,10 +169,10 @@ def run(self, version, token=None):
                 raise RuntimeError(f'Releasing failed: {req.text}')
 
         self.logger.info('Waiting for publishing to finish')
-        for pkg in packages:
+        for pkg in openqa_passed_packages:
             while not self.is_repo_published(pkg['publish_prj'], 'images'):
                 self.logger.debug(f'Waiting for {pkg["publish_prj"]}')
-                time.sleep(20)
+                time.sleep(randint(10, 30))
```

Comment:
IMO either constant times or exponential backoff are ok, but waiting randomly is weird.

Reply:
constant time means access spikes at regular intervals, we should avoid that. exponential means that we need to add an abort condition also. what do you suggest as maximum waiting time?

Reply:
> constant time means access spikes at regular intervals, we should avoid that

AFAIU that applies only if the regularity is at a certain time of day and this can accumulate. In this case the starting point is not that regular, so the checks are not regular either.

(Ideally GoCD gains builtin support for timer jitter like systemd.timer's `RandomizedDelaySec=`)

> exponential means that we need to add an abort condition also. what do you suggest as maximum waiting time?

I'd use an upper limit for the wait time instead and just wait indefinitely if necessary. The alternative is either to fail the job or ignore that publishing didn't complete, which doesn't help either.

Reply:
That said, I do like the simplicity of `randint(10, 30)` compared to some extra lines for linear or exponential backoff with a ceiling... So if regular intervals are an issue then randint is fine for me

Reply:
ok, then lets keep it at randint for now.. otherwise I'll just drop the commit.

Reply:
how about something like

```
for i in range(7):
    if self.is_repo_published(pkg['publish_prj'], 'images'):
        break
    …
    time.sleep(20 * i**2)
```